hexa/pipelines/views.py

```diff
@@ -87,10 +94,94 @@ def credentials(request: HttpRequest) -> HttpResponse:
+    The pipeline ID (`id`) is required, the pipeline's version id (`version`) is optional
+    to run the pipeline with an other version than the latest one.
+
+    In case of JSON payload:
+        the endpoint expects a "config" key containing the pipeline configuration.
+
+    In case of form data:
+        the endpoint expects one key (or multiple instances of the key for multiple parameter) per pipeline parameter.
+
+    User must have the "run_pipeline" permission on the pipeline to be able to run it. The pipeline must be in the same workspace as the request.
+    The current workspace and user's authentication is based on the `Authorization` header.
+    The header must be a Bearer token containing the workspace token (see `<my_workspace>/pipelines` & Create Pipeline)
+
+    Returns:
+        HttpResponse: Returns a dict with the `run_id` key containing the ID of the created run.
+    """
+    if request.user.is_anonymous or getattr(request, "workspace", None) is None:
+        return JsonResponse({"error": "Unauthorized"}, status=401)
+
+    content_type = request.META.get("CONTENT_TYPE")
+    if content_type == "application/x-www-form-urlencoded":
+        data = request.POST
+    elif content_type == "application/json":
+        try:
+            data = json.loads(request.body)
+        except json.JSONDecodeError:
+            return JsonResponse({"error": "Invalid JSON"}, status=400)
+    else:
+        return JsonResponse({"error": "Unsupported content type"}, status=400)
+
+    try:
+        print(request.POST, flush=True)
+        pipeline = (
+            Pipeline.objects.filter_for_user(request.user)
+            .filter(workspace=request.workspace)
+            .get(id=data["id"])
+        )
+        version = pipeline.last_version
+        if "version" in data:
+            version = PipelineVersion.objects.get(
+                pipeline=pipeline, number=data["version"]
+            )
+    except Pipeline.DoesNotExist:
+        return JsonResponse({"error": "Pipeline not found"}, status=404)
+    except PipelineVersion.DoesNotExist:
+        return JsonResponse({"error": "Pipeline version not found"}, status=404)
+
+    # Check if the user can run the pipeline
+    if not request.user.has_perm("pipelines.run_pipeline", pipeline):
+        return JsonResponse({"error": "Forbidden"}, status=403)
+
+    send_mail_notifications = data.get("send_mail_notifications", False)
+
+    # Build the config of the pipeline
+    config = {}
+    if content_type == "application/x-www-form-urlencoded":
+        for parameter in version.parameters:
+            if parameter["code"] not in data:
+                continue
+            if parameter["multiple"]:
+                config[parameter["code"]] = data.getlist(parameter["code"])
+            else:
+                config[parameter["code"]] = data.get(parameter["code"])
+    else:  # We have a JSON payload
+        config = data.get("config", {})
```

Comment:
If we move pipeline id and version to the URL, the `config` key is not necessary anymore and keys could be parameter codes.